buffer/src/main/java/io/netty/buffer/api/LeakInfo.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2021 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.buffer.api;
+
+import io.netty.buffer.api.LeakInfo.TracePoint;
+import io.netty.util.internal.UnstableApi;
+
+import java.util.function.Consumer;
+import java.util.stream.Stream;
+
+/**
+ * Information about a resource leak that happened.
+ * This information is provided to callbacks of the {@link MemoryManager#onLeakDetected(Consumer)} method.
+ */
+@UnstableApi
+public interface LeakInfo extends Iterable<TracePoint> {
+    /**
+     * Create a {@link Stream} of all the {@link TracePoint}s in this {@link LeakInfo}.
+     * The returned {@link Stream} does not need to be closed.
+     *
+     * @return A {@link Stream} of {@link TracePoint}s, covering the life-cycle of the leaked buffer.
```

Comment:
Consider rewording as it is not really tied to buffers.

```suggestion
     * @return A {@link Stream} of {@link TracePoint}s, covering the life-cycle of the leaked resource.
```